testng-eclipse-plugin/src/main/org/testng/eclipse/launch/GroupSelector.java

```diff
@@ -55,16 +58,16 @@
         ; // ignore for the moment
       }
 
-      Object[] projects = new Object[1 + dependencies.length];
+      IJavaProject[] projects = new IJavaProject[1 + dependencies.length];
       projects[0] = selectedProject;
       System.arraycopy(dependencies, 0, projects, 1, dependencies.length);
-      Object[] types = TestSearchEngine.findTests(getCallback().getLaunchConfigurationDialog(),
-          projects, Filters.SINGLE_TEST);
+      Set<IType> types = new HashSet<>();
+      for (IJavaProject project : projects) {
+        types.addAll(Arrays.asList(TestSearchEngine.findTestNGTests(getCallback().getLaunchConfigurationDialog(),
```

Comment:
use the new method `TestSearchEngine.findTestNGTests` which facilitate with JDT search engine